src/sentry/plugins/sentry_mail/activity/assigned.py

```diff
@@ -12,25 +12,36 @@ def get_activity_name(self):
         return 'Assigned'
 
     def get_description(self):
-        # TODO(mattrobenolt): Handle when assignee is Team
-
         activity = self.activity
         data = activity.data
-        if activity.user_id and six.text_type(activity.user_id) == data['assignee']:
-            return u'{author} assigned {an issue} to themselves'
-
-        try:
-            assignee = User.objects.get_from_cache(id=data['assignee'])
-        except User.DoesNotExist:
-            pass
-        else:
-            return u'{author} assigned {an issue} to {assignee}', {
-                'assignee': assignee.get_display_name(),
-            }
-
-        if data.get('assigneeEmail'):
-            return u'{author} assigned {an issue} to {assignee}', {
-                'assignee': data['assigneeEmail'],
-            }
-
-        return u'{author} assigned {an issue} to an unknown user'
+        if data['assigneeType'] == 'user':
+            if activity.user_id and six.text_type(activity.user_id) == data['assignee']:
+                return u'{author} assigned {an issue} to themselves'
+
+            try:
+                assignee = User.objects.get_from_cache(id=data['assignee'])
+            except User.DoesNotExist:
+                pass
+            else:
+                return u'{author} assigned {an issue} to {assignee}', {
+                    'assignee': assignee.get_display_name(),
+                }
+
+            if data.get('assigneeEmail'):
+                return u'{author} assigned {an issue} to {assignee}', {
+                    'assignee': data['assigneeEmail'],
+                }
+
+            return u'{author} assigned {an issue} to an unknown user'
+
+        if data['assigneeType'] == 'team':
+            try:
+                assignee_team = Team.objects.get(id=data['assignee'])
```

Comment:
Can we scope this also to the Organization? Just to make sure we say, don't transfer a Team to another org, then this id is left dangling and now it's possible to query it from another org.

As of right now, this is hypothetical, but we should try always to scope these things by some tenant to prevent issues with leaking data in the future if the behavior were changed.